### PR TITLE
Group Key Management cluster tests enabled.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
Group Key Management cluster tests where disabled due to code generation errors.

#### Change overview
* YAML tests fixed and improved.
* The groupKeyMap attribute of the Group Key Management cluster is now writable.

#### Testing
YAML tests completed successfully.

### CODE REVIEW

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml [hunk 1]
```diff
@@ -41,117 +40,62 @@ tests:
...
-              - name: "groupId"
-                value: 12
+              - name: "groupKeySet"
+                value:
+                    {
+                        groupKeySetID: 101,
+                        securityPolicy: 0,
+                        epochKey0: "\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf",
+                        epochStartTime0: 1110000,
+                        epochKey1: "\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf",
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupKeyManagementCluster.yaml:54]: Shouldn't the values here be `\xb0\xb1.....`, so ephochKey1 is not the same as epochKey2?

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml [hunk 2]
```diff
@@ -179,3 +123,28 @@ tests:
...
+                value: 101
+
+    - label: "KeySet Read (removed)"
+      command: "KeySetRead"
+      arguments:
+          values:
+              - name: "groupKeySetID"
+                value: 101
+      response:
+          error: 0x8B
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupKeyManagementCluster.yaml:141]:
```suggestion
          error: NOT_FOUND
```

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml [hunk 3]
```diff
@@ -179,3 +123,28 @@ tests:
...
+
+    - label: "KeySet Read (removed)"
+      command: "KeySetRead"
+      arguments:
+          values:
+              - name: "groupKeySetID"
+                value: 101
+      response:
+          error: 0x8B
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupKeyManagementCluster.yaml:150]:
```suggestion
          error: NOT_FOUND
```

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml [hunk 4]
```diff
@@ -179,3 +123,28 @@ tests:
...
+
+    - label: "KeySet Read (removed)"
+      command: "KeySetRead"
+      arguments:
+          values:
+              - name: "groupKeySetID"
+                value: 101
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupKeyManagementCluster.yaml:148]: What is the point of doing this same read twice here?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml (hunk 1 above)
Comment (rcasallas-silabs) [reply src/app/tests/suites/TestGroupKeyManagementCluster.yaml:54]: Yes it should. Fixed.

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml (hunk 2 above)
Comment (rcasallas-silabs) [reply src/app/tests/suites/TestGroupKeyManagementCluster.yaml:141]: Suggestion applied.

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml (hunk 4 above)
Comment (rcasallas-silabs) [reply src/app/tests/suites/TestGroupKeyManagementCluster.yaml:148]: Not sure why this test got duplicated. Removed.

File: src/app/tests/suites/TestGroupKeyManagementCluster.yaml (hunk 3 above)
Comment (rcasallas-silabs) [reply src/app/tests/suites/TestGroupKeyManagementCluster.yaml:150]: Suggestion applied.

**Pull request closed without merging**